life expectancy.Rmd

```diff
@@ -8,50 +8,51 @@ knitr::opts_chunk$set(echo = TRUE)
 ```
 
 
-The data I am working on is about the life expectancy from all contries from year 1800 to 2100, 
+
+### The data I worked on is of the life expectancy from all contries from year 1800 to 2100, from this website:
 [Behavioral Risk Factor Surveillance System (BRFSS)](https://www.cdc.gov/brfss/about/index.htm)
 
 ```{r message = FALSE,warning = FALSE}
 library(tidyverse)
 library(ggplot2)
 ```
 
-Import and take a look at the data
+#### Import and take a look at the data
 
 ```{r}
 life <- read.csv("life_expectancy_years.csv", header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)
 head(life)
 ```
 
-Modify the dataset structure
+#### Modify the dataset structure
 ```{r}
 rownames(life) <- life[,1]
 life$country <- NULL
 ```
 
-Then, I would like to look at the life expactancy trend for China from year 1800 to 2100
+#### Then, I would like to look at the life expactancy trend for China from year 1800 to 2100
 
 ```{r}
 year <- colnames(life)
 year <- as.numeric(year)
 China <- as.numeric(life[c("China"),])
 ```
 
-Then, I would like to look at the life expactancy trend for United States from year 1800 to 2100
+#### Then, I would like to look at the life expactancy trend for United States from year 1800 to 2100
 
 ```{r}
 year <- colnames(life)
 year <- as.numeric(year)
 USA <- as.numeric(life[c("United States"),])
 ```
 
-Combine the subsets into a new table, and esier to draw plots from
```

Comment:
It seems like getting the data formatted in a way that is conducive to drawing plots is half the battle in R! I have often heard from other people that it is easiest to clean and format your data in excel first, so I commend you for doing it all in R! I tried but could not easily achieve this myself in R, so I took the easy way out and just did it in excel!

Reply:
Yes, I totally agree! It is easier to clean up in Excel first!
(In some cases, some people, especially the computer experts may find this way not "professional" enough, they would prefer the code to do everything, avoid "hard-coding", although I don't think clean up the data in Excel is really "hard-coding", but yes, for our daily purposes, a clean Excel def helps a lot, and save much time!)